src/components/adminWaiverEdit.vue

```diff
@@ -0,0 +1,395 @@
+          this.waiver.header = data['headers'];
+          this.waiver.items = data['items'];
+          this.waiver.footer = data['footer'];
+
+          // init invalid list for each header
+          this.waiver.header.forEach(h => {
+            this.invalid.header.push(false);
+          });
+
+          // init invalid list for each item
+          this.waiver.items.forEach(i => {
+            this.invalid.items.push(this.invalid_item);
+          });
+          setTimeout(x => {this.matchHeight()}, 2000);
+        })
+      },
+
+      matchHeight() {
+        let preview_height = this.$refs.updateWaiver.scrollHeight - 50;
+        this.preview_btn_styles = {'height': preview_height + 'px'};
+      },
+
+      scrollToPreview() {
+        let element = '#waiver_preview';
+        let duration = 2000;
+        var VueScrollTo = require('vue-scrollto');
+        VueScrollTo.scrollTo(element, duration)
+      },
+
+      deleteItem(item_idx) {
+        // remove the item from the list
+        this.waiver.items.splice(item_idx, 1);
+      },
+
+      addItem() {
+        let count = this.waiver.items.length;
+        // add a new item object to the list
+        this.waiver.items.push(
+          {
+            'order': count + 1,
+            'title': '',
+            'text': '',
+            'initials': true
+          }
+        );
+      },
+
+      cancelSubmission(event) {
+        if (event) { event.preventDefault(); }
+        this.getWaiver(this.type);
+      },
+
+      headerValidator: function(invalid, idx) {
+        this.invalid.header.splice(idx, 1, invalid);
+        this.headers_have_errors = this.checkInvalidList(this.invalid.header);
+      },
+
+      itemValidator: function(invalid, idx, key) {
+        let item_state = this.invalid.items[idx];
+        item_state[key] = invalid;
+        this.invalid.items.splice(idx, 1, item_state);
+        this.items_have_errors = this.checkInvalidList(this.invalid.items);
+      },
+
+      checkInvalidList: function(list) {
+        let invalid = false;
+        list.forEach(i => {
+          if (this.isDict(i)) {
+            for (var key in i) {
+              if (i[key]) { invalid = true; }
+            }
+          }
+          else if (i) { invalid = true; }
+        });
+        return invalid;
+      },
+
+      isDict: function(x) {
+        return typeof x==='object' && x!==null && !(x instanceof Array) && !(x instanceof Date);
+      },
+
+      update_waiver() {
+        let dataObj = {
+          'title': this.waiver.title,
+          'headers': this.waiver.header,
+          'items': this.waiver.items,
+          'footer': this.waiver.footer
+        }
+        localforage.getItem('X_TOKEN')
+        .then(session => {
+          axios.put('/api/v1/admin/waiver/' + this.type + '/update', {
+            headers: { 'x-token': session },
+            data: dataObj
+          })
+          .then(res => {
+            this.$router.push('/admin/applications');
+          })
```

Comment:
do you want to send them to this route? or send them back to the same page with a success message?

Reply:
a success message would be preferable - I'll add that before the next push